jerry-ext/handler/handler-assert.c

```diff
@@ -18,30 +18,27 @@
 
 #include <inttypes.h>
 
+typedef enum
+{
+  JERRYX_OP_EQUAL = 0,
+  JERRYX_OP_STRICT_EQUAL = 1,
+  JERRYX_OP_NOT_EQUAL = 2,
+  JERRYX_OP_NOT_STRICT_EQUAL = 3
+} operations;
 /**
- * Hard assert for scripts. The routine calls jerry_port_fatal on assertion failure.
- *
- * Notes:
- *  * If the `JERRY_FEATURE_LINE_INFO` runtime feature is enabled (build option: `JERRY_LINE_INFO`)
- *    a backtrace is also printed out.
- *
- * @return true - if only one argument was passed and that argument was a boolean true.
- *         Note that the function does not return otherwise.
+ * TODO
  */
-jerry_value_t
-jerryx_handler_assert_fatal (const jerry_value_t func_obj_val, /**< function object */
-                             const jerry_value_t this_p, /**< this arg */
-                             const jerry_value_t args_p[], /**< function arguments */
-                             const jerry_length_t args_cnt) /**< number of function arguments */
+static void
+jerryx_fatal (const jerry_value_t *message)
 {
-  (void) func_obj_val; /* unused */
-  (void) this_p; /* unused */
-
-  if (args_cnt == 1
-      && jerry_value_is_boolean (args_p[0])
-      && jerry_get_boolean_value (args_p[0]))
+  if (message != NULL && jerry_value_is_string (*message))
   {
-    return jerry_create_boolean (true);
+    jerry_char_t buff[256];
+    jerry_size_t buf_size = jerry_string_to_char_buffer (*message, buff, sizeof (buff));
```

Comment:
What if the message is longer than 256 bytes? IMO there is two ways to go:
 - 1. Use `https://jerryscript.net/api-reference/#jerry_substring_to_char_buffer` with a while loop and a fix size buffer.
 - 2. Query the string size, allocate a buffer for it, print the content of the buffer, than release the buffer.

I'd prefer the `1`st way, since it's less memory intensive.